---
int8 headingSource
bool success
